openeo/extra/job_management/__init__.py

```diff
@@ -668,25 +708,72 @@ def on_job_cancel(self, job: BatchJob, row):
         """
         pass
 
+    def _postprocess_futures(self, worker_pool_results, df, stats):
+        """
+        Processes completed tasks from the worker thread pool and updates job statuses in the job dataframe.
+
+        This method iterates over the results of completed tasks from the worker pool (e.g., job start attempts).
+        For each task, it updates the job's status in the DataFrame based on whether the start was succesful
+        and the job moved to the queued status or wheter the job start failed.
+
+        :param pool_output:
+            A list of tuples, where each tuple contains:
+                - task: An instance of `_JobStartTask` representing the job start task.
+                - result: A tuple containing:
+                    - job_id (str): The ID of the job associated with the task.
+                    - success (bool): Whether the task succeeded.
+                    - data (str): the resulting status or an error message.
+
+        :param df:
+            The dataFrame that specifies the jobs and tracks their statuses. The status of jobs in the DataFrame is updated based on the task results.
+
+        :param stats:
+            The dictionary for tracking statistics related to job processing. The statistics are updated based on the task results.
+
+        """
+
+        for task, result in worker_pool_results:
+            job_id, success, data = result
```

Comment:
as mentioned I would decouple this ad-hoc tuple unpacking by using a `_TaskResult` instead